docs/writing-rules/data-flow.md

```diff
@@ -1,38 +1,60 @@
 ---
 id: data-flow
 description: >-
-  Learn how to use Semgrep’s intuitive syntax to write rules specific to your codebase. You can write and share rules directly from your browser using the Semgrep Playground, or write rules in your terminal and run them on the command line.
+  Learn about Semgrep’s data-flow analyses.
 ---
 
 # Data-flow analysis
 
-Semgrep can perform intra-procedural flow-sensitive analyses. The data-flow engine still has several limitations, therefore expect both false positives and false negatives. False positives could be removed by using [pattern-not](../writing-rules/rule-syntax.md#pattern-not).
-
-A non-exhaustive list of current limitations:
-
-- The analyses are not aware of _aliasing_.
-- The analyses do not track individual elements in data structures, although there is limited support for record fields.
-- `break`, `continue`, and `switch` statements are not properly handled yet.
-- `try-catch-finally` is only partially supported, not all possible execution paths are considered.
-
-Data-flow analysis is used for [constant propagation](#constant-propagation).
+Semgrep comes with an intra-procedural data-flow analysis engine that powers [constant propagation](#constant-propagation) and [taint tracking](#taint-tracking). (At present you cannot write your own data-flow analyses.) Make sure that you understand the [limitations](#limitations) of the data-flow engine.
 
 ## Constant propagation
 
-Semgrep supports intra-procedural constant propagation. This tracks whether a variable must carry a constant value at each point in the program.
+Semgrep supports intra-procedural constant propagation. This analysis tracks whether a variable _must_ carry a constant value at a given point in the program. Semgrep then performs constant folding when matching literal patterns. For now we can track Boolean, numeric, and string constants.
```

Comment:
```suggestion
Semgrep supports intra-procedural constant propagation. This analysis tracks whether a variable _must_ carry a constant value at a given point in the program. Semgrep then performs constant folding when matching literal patterns. For now it can track Boolean, numeric, and string constants.
```